xds/server.go

```diff
@@ -93,12 +93,11 @@ func NewGRPCServer(opts ...grpc.ServerOption) (*GRPCServer, error) {
 	// simplifies the code by eliminating the need for a mutex to protect the
 	// xdsC and xdsClientClose fields.
 	newXDSClient := newXDSClient
```

Comment:
Instead of having this `newXDSClient` var which can be overridden by tests to customize xDS client creation, the xDS server (and possibly the xDS resolver) should contain an `xdsClientPool` as a global var which would be set to the default pool at `init` time and can be customized by tests.